compose/config/interpolation.py

```diff
@@ -26,9 +29,46 @@ def process_item(name, config_dict):
     )
 
 
-def interpolate_value(name, config_key, value, section, mapping):
+def _cast_interpolated(interpolated_value, expected_type):
+
+    if expected_type == "string":
+        return None
+    elif expected_type == "number":
+        try:
+            return int(interpolated_value)
+        except ValueError:
+            return None
+    elif expected_type == "boolean":
+        if "true" == interpolated_value.lower():
+            return True
+        if "false" == interpolated_value.lower():
+            return False
+        return None
+    elif expected_type == "array":
+        if isinstance(interpolated_value, list):
+            return interpolated_value
+        return json.loads("{{\"data\":{0}}}".format(interpolated_value.replace("'", "\"")))["data"]
+
```

Comment:
As I said I have not written python for years, but it seems that `json.loads` does not work with an Array.
Looks like I am not the only one to face the issue 
=> http://stackoverflow.com/questions/10973614/convert-json-array-to-python-list

However, it may be just because of issue between simple quotes (') and double quotes (")